### PR TITLE
bump vendor/postgres to fix walproposer segfault

### PR DESCRIPTION
**Pull request by arssher**

### CODE REVIEW

**Review by sharnoff — APPROVED** [review]

**Review by hlinnaka — APPROVED** [review]
Let's get this in, to fix the CI failures.

I find it hard to follow the logic in walproposer.c, It could use some  further refactoring, commenting, etc. to make it easier to understand. But that's a bigger, pre-existing issue, not this PR's fault.

**Comment by arssher** [comment]
Per offline review from Konstanin fix could be simpler, but ok, let's fix CI first and do that separately.
